assets/js/src/wizard/woocommerce_import_controller.jsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import MailPoet from 'mailpoet';
+import WelcomeWizardHeader from './header.jsx';
+import WooCommerceImportListStep from './steps/woo_commerce_import_list_step.jsx';
+
+class WooCommerceImportController extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: false,
+    };
+    this.updateSettings = this.updateSettings.bind(this);
+    this.scheduleImport = this.scheduleImport.bind(this);
+    this.finishWizard = this.finishWizard.bind(this);
+    this.submit = this.submit.bind(this);
+  }
+
+  finishWizard() {
+    this.setState({ loading: true });
+    window.location = window.finish_wizard_url;
+  }
+
+  updateSettings(data) {
+    return MailPoet.Ajax.post({
+      api_version: window.mailpoet_api_version,
+      endpoint: 'settings',
+      action: 'set',
+      data,
+    }).fail((response) => {
+      this.setState({ loading: false });
+      if (response.errors.length > 0) {
```

Comment:
I think this can happen in case some network error during the request and we should check if the response itself is available. I would add fallback to `Unknown error` message for such strange usecases. What do you think?